### PR TITLE
[FIX] endpoint_route_handler: Rollback dedicated routing cursor

### PR DESCRIPTION
**Pull request by paradoxxxzero**
Unless I am missing something, we should not commit this transaction (`registry.cursor()` commits at [`__exit__ `](https://github.com/odoo/odoo/blob/16.0/odoo/sql_db.py#L176)).

This fix the failing tests in rest-framework: https://github.com/OCA/rest-framework/issues/470

### CODE REVIEW

**Review by simahawk — APPROVED** [review]
LG, good catch :)

File: endpoint_route_handler/models/ir_http.py [hunk 1]
```diff
@@ -71,6 +72,10 @@ def routing_map(cls, key=None):
                 cls._routing_map = {}
                 cls._rewrite_len = {}
                 cls._endpoint_route_last_version = last_version
+        finally:
+            cr.rollback()
```
Comment (simahawk) [inline endpoint_route_handler/models/ir_http.py:79]: can you add a comment here pls?
Mention that the new cr is only to bypass transaction isolation during a read  or smth similar.

Comment (paradoxxxzero) [reply endpoint_route_handler/models/ir_http.py:79]: Done

**Review by JordiMForgeFlow — APPROVED** [review]

**Review by lmignon — APPROVED** [review]
Thank you for the fix @paradoxxxzero LGTM

**Pull request merged**